server/src/test/java/org/opensearch/action/search/AbstractSearchAsyncActionTests.java

```diff
@@ -695,6 +677,13 @@ public void testOnPhaseListenersWithDfsType() throws InterruptedException {
         assertThat(testListener.getPhaseMetric(searchDfsQueryThenFetchAsyncAction.getSearchPhaseName()), greaterThanOrEqualTo(delay));
         assertEquals(1, testListener.getPhaseTotal(searchDfsQueryThenFetchAsyncAction.getSearchPhaseName()));
         assertEquals(0, testListener.getPhaseCurrent(searchDfsQueryThenFetchAsyncAction.getSearchPhaseName()));
+        assertingListener.onPhaseEnd(
```

Comment:
We should not be doing such calls, right?

Reply:
We need to simulate phase end callback in sendSearchResponse() for these cases. I have moved it into run() function which may make more sense.